### ISSUE TITLE
GlobalSingleton matching query does not exist.

### ISSUE THREAD

**Issue by PhoenixAlx** [oTree-org/otree-core#443]
Hi,

I have a problem. When I am going to sessions from Menu, i have the next error:

Django Version: 	1.8.8
Exception Type: 	DoesNotExist
Exception Value: 	

GlobalSingleton matching query does not exist.

Exception Location: 	/usr/lib/python3.5/site-packages/django/db/models/query.py in get, line 334

This starte to happen from that I was to remove various sessions of dictator game.

Can you help me? Please

**Comment by leliel12** [comment]
can you share the full stack of error?

**Comment by PhoenixAlx** [comment]
This? @leliel12 



Django Version: 1.8.8
Python Version: 3.5.1
Installed Applications:
['django.contrib.auth',
 'otree',
 'floppyforms',
 'django.contrib.admin',
 'django.contrib.contenttypes',
 'django.contrib.sessions',
 'django.contrib.messages',
 'django.contrib.staticfiles',
 'otree.models_concrete',
 'otree.timeout',
 'djcelery',
 'kombu.transport.django',
 'rest_framework',
 'sslserver',
 'idmap',
 'corsheaders',
 'survey',
 'trust',
 'prisoner',
 'payment_info',
 'bargaining',
 'matching_pennies_tutorial',
 'cournot_competition',
 'ultimatum',
 'beauty',
 'real_effort',
 'lemon_market',
 'principal_agent',
 'volunteer_dilemma',
 'matching_pennies',
 'public_goods_simple',
 'trust_simple',
 'stackelberg_competition',
 'traveler_dilemma',
 'stag_hunt',
 'battle_of_the_sexes',
 'bertrand_competition',
 'vickrey_auction',
 'dictator',
 'public_goods',
 'common_value_auction']
Installed Middleware:
['otree.middleware.CheckDBMiddleware',
 'otree.middleware.HumanErrorMiddleware',
 'corsheaders.middleware.CorsMiddleware',
 'django.contrib.sessions.middleware.SessionMiddleware',
 'django.middleware.common.CommonMiddleware',
 'django.middleware.csrf.CsrfViewMiddleware',
 'django.contrib.auth.middleware.AuthenticationMiddleware',
 'django.contrib.messages.middleware.MessageMiddleware']


Traceback:
File "/usr/lib/python3.5/site-packages/django/core/handlers/base.py" in get_response
  138.                         response = middleware_method(request, e)
File "/usr/lib/python3.5/site-packages/otree/middleware.py" in process_exception
  32.         common_internal.reraise(exception)
File "/usr/lib/python3.5/site-packages/otree/common_internal.py" in reraise
  317.         six.reraise(original_cls, original, sys.exc_info()[2])
File "/usr/lib/python3.5/site-packages/six.py" in reraise
  686.         raise value
File "/usr/lib/python3.5/site-packages/django/core/handlers/base.py" in get_response
  132.                     response = wrapped_callback(request, *callback_args, **callback_kwargs)
File "/usr/lib/python3.5/site-packages/django/views/generic/base.py" in view
  71.             return self.dispatch(request, *args, **kwargs)
File "/usr/lib/python3.5/site-packages/django/views/generic/base.py" in dispatch
  89.         return handler(request, *args, **kwargs)
File "/usr/lib/python3.5/site-packages/vanilla/model_views.py" in get
  231.                 paginator=None,
File "/usr/lib/python3.5/site-packages/otree/views/admin.py" in get_context_data
  898.         global_singleton = GlobalSingleton.objects.get()
File "/usr/lib/python3.5/site-packages/django/db/models/manager.py" in manager_method
  127.                 return getattr(self.get_queryset(), name)(*args, **kwargs)
File "/usr/lib/python3.5/site-packages/idmap/queryset.py" in get
  37.             instance = super(SharedMemoryQuerySet, self).get(**kwargs)
File "/usr/lib/python3.5/site-packages/django/db/models/query.py" in get
  334.                 self.model._meta.object_name

Exception Type: DoesNotExist at /sessions/
Exception Value: GlobalSingleton matching query does not exist.

**Comment by oTree-org** [comment]
I can't reproduce this -- is it occurring consistently for you?

**Comment by leliel12** [comment]
i have the same issue. Do you run resetdb before start the server?

On Thu, 10 Mar 2016 at 01:05 oTree-org <notifications@github.com> wrote:

> I can't reproduce this -- is it occurring consistently for you?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/oTree-org/otree-core/issues/443#issuecomment-194653676>
> .
>
-- 

Juan BC (from phone)

**Comment by PhoenixAlx** [comment]
@leliel12 @oTree-org yes, both.
I will try resetdb again.

**Comment by PhoenixAlx** [comment]
ok.
I have done resetdb and all ok. But after...
I do new session-->dictator game, 2 players.
Click in menu sessions and I see that session. 
Click in default button. 
Click in select all sessions and delete.
Then that error appears

**Comment by oTree-org** [comment]
Yes I understand the issue now. You made a session the default and then deleted the default session. Anyway, in our next release oTree we will significantly change this "default session" feature; hopefully it will not have this issue.

Temporary fix is to run "otree shell" and enter these commands:

```
from otree.models.session import GlobalSingleton
GlobalSingleton.objects.get_or_create()
```

**Comment by oTree-org** [comment]
Fixed in 0.5
